src/features/lessons/list-of-lessons/list-of-lessons.js

```diff
@@ -65,18 +63,58 @@ export const ListOfLessons = () => {
     setVisibleLessonsList(filteredLessonsList.slice(indexOfFirst, indexOfLast));
   }, [currentPage, filteredLessonsList]);
 
-  const handleSearchTheme = (inputValue) => setSearchLessonsThemeValue(inputValue);
+  function getYearMonthDayFormat(inputFormat) {
+    function pad(s) { return (s < 10) ? '0' + s : s; }
+    const d = new Date(inputFormat)
+    return [pad(d.getFullYear()), pad(d.getMonth()+1), d.getDate(), ].join('-')
+  }
 
-  const handleSearchDate = (event) => setSearchLessonsDateValue(event.target.value);
+  function getMonthDayYearFormat(inputFormat) {
+    function pad(s) { return (s < 10) ? '0' + s : s; }
+    const d = new Date(inputFormat)
+    return [pad(d.getMonth()+1), pad(d.getDate()), d.getFullYear()].join('-')
+  }
+
+  const currentDateString = getYearMonthDayFormat(new Date());
+  const halfMonthDays = 15;
+  const halfMonthPastDate = new Date();
+  halfMonthPastDate.setDate(halfMonthPastDate.getDate() - halfMonthDays);
+  const halfMonthPastDateString = getYearMonthDayFormat(halfMonthPastDate);
```

Comment:
Why wouldn't you use commonHelpers.transformDateTime({}).reverseDate to get the same result?

Reply:
fixed with new commit